nodejs/awsx/apigateway/api.ts

```diff
@@ -986,11 +1012,31 @@ function addIntegrationRouteToSwaggerSpec(
     const swaggerPath = route.path.endsWith("/") ? route.path : route.path + "/";
     const swaggerPathProxy = swaggerPath + "{proxy+}";
 
-    addSwaggerOperation(swagger, swaggerPath, method,
-        createSwaggerOperationForProxy(target, /*useProxyPathParameter:*/ false));
+    const swaggerOpWithoutProxyPathParam = createSwaggerOperationForProxy(target, /*useProxyPathParameter:*/ false);
+    const swaggerOpWithProxyPathParam = createSwaggerOperationForProxy(target, /*useProxyPathParameter:*/ true);
+
+    if (route.authorizers) {
+        const authRecords = addAuthorizersToSwagger(swagger, route.authorizers, apiAuthorizers);
+        addAuthorizersToSwaggerOperation(swaggerOpWithoutProxyPathParam, authRecords);
+        addAuthorizersToSwaggerOperation(swaggerOpWithProxyPathParam, authRecords);
+    }
+    if (route.requiredParameters) {
+        addRequiredParametersToSwaggerOperation(swaggerOpWithoutProxyPathParam, route.requiredParameters);
+        addRequiredParametersToSwaggerOperation(swaggerOpWithProxyPathParam, route.requiredParameters);
+    }
+    if (route.requestValidator) {
+        swaggerOpWithoutProxyPathParam["x-amazon-apigateway-request-validator"] = route.requestValidator;
+        swaggerOpWithProxyPathParam["x-amazon-apigateway-request-validator"] = route.requestValidator;
+    }
+    if (route.apiKeyRequired) {
+        addAPIkeyToSecurityDefinitions(swagger);
+        addAPIKeyToSwaggerOperation(swaggerOpWithoutProxyPathParam);
+        addAPIKeyToSwaggerOperation(swaggerOpWithProxyPathParam);
+    }
```

Comment:
not possible to extract a function that will do the 4 steps, but then call it for both swaggerOpWithoutProxyPathParam and swaggerOpWithProxyPathParam?

Reply:
feels like we've triplicated this with addEventHandlerRouteToSwaggerSpec and addStaticRouteToSwaggerSpec (Esp for the file/dir handling).  This feels like this must be extractable.